impl/src/attr.rs

```diff
@@ -115,6 +117,30 @@ pub fn get(input: &[Attribute]) -> Result<Attrs> {
                 original: attr,
                 span,
             });
+        } else if attr.path().is_ident("allow") {
+            // Save `allow`s in order to put them on all `impl`s of the decorated type,
+            // similarly to what the Debug derive macro does.
+            //
+            // More specifically, this code:
+            // ```rust
+            // #[allow(deprecated, never_type_fallback_flowing_into_unsafe)]
+            // #[derive(Debug)]
+            // pub struct B;
+            // ```
+            // gets expanded into:
+            // ```rust
+            // #[allow(deprecated, never_type_fallback_flowing_into_unsafe)]
+            // pub struct B;
+            // #[automatically_derived]
+            // #[allow(deprecated, never_type_fallback_flowing_into_unsafe)]
+            // impl ::core::fmt::Debug for B {
+            //     #[inline]
+            //     fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
+            //         ::core::fmt::Formatter::write_str(f, "B")
+            //     }
+            // }
+            // ```
+            attrs.allows = Some(attr);
```

Comment:
This would drop all but the last `allow` attribute. When there are multiple attributes, the expected behavior would be that they all make it to the impl, not just the last one.

Reply:
You're right; the only scenario I considered was and tested was with all `allow`s listed in one attribute.